rewrite-java/src/main/java/org/openrewrite/java/ChangeStaticFieldToMethod.java

```diff
@@ -0,0 +1,118 @@
+
+    @Option(displayName = "Old field name",
+            description = "The simple name of the static field to replace.",
+            example = "EMPTY_LIST")
+    String oldFieldName;
+
+    @Option(displayName = "New class name",
+            description = "The fully qualified name of the class containing the method to use. Leave empty to keep the same class.",
+            example = "java.util.List",
+            required = false)
+    @Nullable
+    String newClassName;
+
+    @Option(displayName = "New method name",
+            description = "The simple name of the method to use. The method must be static and have no arguments.",
+            example = "of")
+    String newMethodName;
+
+    @Override
+    public String getDisplayName() {
+        return "Change static field access to static method access";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Migrate accesses to a static field to invocations of a static method";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesField<>(oldClassName, oldFieldName);
+    }
+
+    @Override
+    protected JavaVisitor<ExecutionContext> getVisitor() {
+        return new JavaVisitor<ExecutionContext>() {
+            @Override
+            public J visitFieldAccess(J.FieldAccess fieldAccess, ExecutionContext ctx) {
+                if (getCursor().firstEnclosing(J.Import.class) == null &&
+                        TypeUtils.isOfClassType(fieldAccess.getTarget().getType(), oldClassName) &&
+                        fieldAccess.getSimpleName().equals(oldFieldName)) {
+                    return useSystemLineSeparator(fieldAccess);
+                }
+                return super.visitFieldAccess(fieldAccess, ctx);
+            }
+
+            @Override
+            public J visitIdentifier(J.Identifier ident, ExecutionContext executionContext) {
+                JavaType.Variable varType = TypeUtils.asVariable(ident.getFieldType());
+                if (varType != null &&
+                        TypeUtils.isOfClassType(varType.getOwner(), oldClassName) &&
+                        varType.getName().equals(oldFieldName)) {
+                    return useSystemLineSeparator(ident);
+                }
+                return ident;
+            }
+
+            @NotNull
+            private J useSystemLineSeparator(J j) {
+                final String newClass = newClassName == null ? oldClassName : newClassName;
+
+                maybeRemoveImport(oldClassName);
+                // We don't need imports for java.lang - maybe AddImport should know that?
+                if (!newClass.startsWith("java.lang.")) {
+                    maybeAddImport(newClass);
+                }
+                final String simpleClassName = StringUtils.substringAfterLast(newClass, ".");
+                final String methodInvocationTemplate = simpleClassName + "." + newMethodName + "()";
```

Comment:
Currently, the tests do not succeed due to a missing import statement.

My debugging showed that `AddImport` uses `FindTypes.find` which does not recognize the reference to the class.